src/mscorlib/src/System/Guid.cs

```diff
@@ -588,307 +663,287 @@ private static bool TryParseGuid(String g, GuidStyles flags, ref GuidResult resu
             result.parsedGuid._k = bytes[7];
 
             // Check for last '}'
-            if(numStart+numLen+1 >= guidString.Length || guidString[numStart+numLen+1] != '}') {
+            if (numEnd + 1 >= length || guidString[numEnd + 1] != '}')
+            {
                 result.SetFailure(ParseFailureKind.Format, "Format_GuidEndBrace");
                 return false;
             }
 
             // Check if we have extra characters at the end
-            if( numStart+numLen+1 != guidString.Length -1) {
+            if (numEnd + 1 != length - 1)
+            {
                 result.SetFailure(ParseFailureKind.Format, "Format_ExtraJunkAtEnd");
                 return false;
             }
-            
+
             return true;
         }
-        
-        // Check if it's of the form dddddddddddddddddddddddddddddddd
-        private static bool TryParseGuidWithNoStyle(String guidString, ref GuidResult result) {
-            int startPos=0;
-            int temp;
-            long templ;
-            int currentPos = 0;
 
-            if(guidString.Length != 32) {
-                result.SetFailure(ParseFailureKind.Format, "Format_GuidInvLen");
-                return false;
+        private static unsafe char* EatAllWhitespace(char* guidString, ref int start, ref int length)
+        {
+            int whitespaceCount = 0;
+            for (int i = start; i < start + length; i++)
+            {
+                if (char.IsWhiteSpace(guidString[i]))
+                {
+                    whitespaceCount++;
+                }
             }
 
-            for(int i= 0; i< guidString.Length; i++) {
-                char ch = guidString[i];
-                if(ch >= '0' && ch <= '9') {
-                    continue;
-                }
-                else {
-                    char upperCaseCh = Char.ToUpper(ch, CultureInfo.InvariantCulture);
-                    if(upperCaseCh >= 'A' && upperCaseCh <= 'F') {
-                        continue;
-                    }
+            if (whitespaceCount == 0)
+            {
+                // No whitespace found, don't allocate a char[] array or string
+                return guidString;
+            }
+
+            int newLength = 0;
+            char[] chars = new char[length - whitespaceCount];
+            for (int i = start; i < start + length; i++)
+            {
+                char c = guidString[i];
+                if (!char.IsWhiteSpace(c))
+                {
+                    chars[newLength++] = c;
                 }
+            }
+            start = 0;
+            length = newLength;
+            fixed (char* newGuidString = new string(chars, 0, newLength))
+            {
+                return newGuidString;
```

Comment:
I believe this is unsafe, after the fixed block exits, there's nothing preventing the GC from relocating the managed object backing `newGuidString` and preventing the pointer from dangling.